extensions/builder.md

```diff
@@ -0,0 +1,145 @@
+# Builder
+
+A **builder** is an image that MUST contain an implementation of the lifecycle, and build-time environment, and MAY contain buildpacks. Platforms SHOULD use builders to ease the build process.
```

Comment:
It seems possible for a user to want a builder to just be the execution environment, without any explicit buildpacks (those could be provided with `--buildpack` to `pack`). Is that something we would want to support, or should I make this "MUST contain buildpack blobs" as well?

Reply:
We should definitely allow builders that do not contain buildpacks.

Reply:
557b34f